presto-thrift-connector/src/main/java/com/facebook/presto/connector/thrift/clientproviders/RetryingPrestoThriftServiceProvider.java

```diff
@@ -58,6 +62,7 @@ public RetryingPrestoThriftServiceProvider(@NonRetrying PrestoThriftServiceProvi
         requireNonNull(retryExecutor, "retryExecutor is null");
         requireNonNull(config, "config is null");
 
+        retryingPrestoThriftServiceStats = new RetryingPrestoThriftServiceStats();
```

Comment:
Initialize in the field declaration. Alternatively, you could inject this and export it directly, removing the need for the method below.